packages/jest-config/src/resolveConfigPath.ts

```diff
@@ -57,16 +57,21 @@ const resolveConfigPathByTraversing = (
   initialPath: Config.Path,
   cwd: Config.Path,
 ): Config.Path => {
-  const jestConfig = JEST_CONFIG_EXT_ORDER.map(ext =>
+  const configFiles = JEST_CONFIG_EXT_ORDER.map(ext =>
     path.resolve(pathToResolve, getConfigFilename(ext)),
-  ).find(isFile);
-  if (jestConfig) {
-    return jestConfig;
+  ).filter(isFile);
+
+  const packageJson = findPackageJson(pathToResolve);
+  if (packageJson && hasPackageJsonJestKey(packageJson)) {
+    configFiles.push(packageJson);
+  }
+
+  if (configFiles.length > 1) {
+    throw new Error(makeMultipleConfigsError(configFiles));
```

Comment:
Sounds reasonable.

Should I hide it behind a flag? Or just delete it and rewrite it for warnings?

Reply:
I'd do `console.warn` instead of throwing, make the unit tests mock out the console, while the e2e test is essentially the same except for the exit code

Reply:
`resolveConfigPath` has to be a pure function (no `console.warn` there), because it is run multiple times.

There is `hasDeprecationWarnings: boolean` flag somewhere. I will look into it later.

Reply:
`hasDeprecationWarnings` has no text associated with it.

The best solution I found was to propagate `shouldLogWarnings`. But I'm not sure it is good enough.

Reply:
why is it called multiple times? that seems odd...

Reply:
The "second" usage is here: https://github.com/facebook/jest/blob/a22ed6592f772df6b19314d9f06a7285c3b0968b/packages/jest-config/src/index.ts#L287

Just to check if there are multiple projects defined in the config (I'm not sure what that is).

Reply:
@SimenB ^ this one is still not solved.

Reply:
gave that a whack with [`6a52634` (#11922)](https://github.com/facebook/jest/pull/11922/commits/6a5263449fe96c11aba9b23a2a85252ac35a279e)

Reply:
Ok, so you went with `shouldLogWarnings`. :+1: